tests/monotouch-test/Network/NWParametersTest.cs

```diff
@@ -0,0 +1,433 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+#if XAMCORE_2_0
+using CoreFoundation;
+using Foundation;
+using Network;
+using ObjCRuntime;
+using Security;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.Network;
+using MonoTouch.Security;
+#endif
+
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.Network {
+
+	public class NWParametersTest {
+
+		AutoResetEvent secureEvent;  // used to let us know the handler was indeed called.
+		AutoResetEvent configureEvent;  // used to let us know the handler was indeed called.
+		AutoResetEvent connectedEvent;  // used to let us know when the connection was established so that we can access the NWPath 
+		bool secureConnectionWasSet = false;
+		bool protocolConfigured = false;
+		List<NWInterface> interfaces = new List<NWInterface> ();
+
+		[SetUp]
+		public void SetUp ()
+		{
+			secureEvent = new AutoResetEvent(false);
+			configureEvent = new AutoResetEvent(false);
+			connectedEvent = new AutoResetEvent(false);
+			secureConnectionWasSet = false;
+			protocolConfigured = false;
+			interfaces = new List<NWInterface> ();
+			// we create a connection which we are going to use to get the availabe
+			// interfaces, that way we can later test protperties of the NWParameters class.
+			using (var parameters = NWParameters.CreateUdp ())
+			using (var endpoint = NWEndpoint.Create ("wwww.google.com", "80"))
+			using (var connection = new NWConnection (endpoint, parameters)) {
+				connection.SetStateChangeHandler (ConnectionStateHandler);
+				connection.SetQueue (DispatchQueue.MainQueue);
+				connection.Start ();
+				connectedEvent.WaitOne (500);
+				using (var path = connection.CurrentPath)
+				{
+					path.EnumerateInterfaces (EnumerateInterfacesHandler);
+				}
+			}
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			secureEvent = null;
+			secureConnectionWasSet = false;
+			protocolConfigured = false;
+			foreach (var i in interfaces) {
+				i.Dispose ();
+			}
+		}
+
+		void ConnectionStateHandler (NWConnectionState state, NWError error)
+		{
+			var errno = (SslStatus)(error != null ? error.ErrorCode : 0);
+			switch (state){
+			case NWConnectionState.Ready:
+				connectedEvent.Set ();
+				break;
+			}
+		}
+
+		void EnumerateInterfacesHandler (NWInterface nwInterface)
+		{
+			interfaces.Add (nwInterface);
+		}
+
+		Action<NWProtocolOptions> CreateTlsHandler ()
+		{
+			return (NWProtocolOptions options) => {
+				secureConnectionWasSet = true;
+				secureEvent.Set ();
+			};
+		}
+
+		Action<NWProtocolOptions> CreateConfigureProtocolHandler ()
+		{
+			return (NWProtocolOptions options) => {
+				protocolConfigured = true;
+				configureEvent.Set ();
+			};
+		}
+
+		[Test]
+		public void CreateSecureUpdTest ()
+		{
+			TestRuntime.AssertSystemVersion (PlatformName.iOS, 12, 0, throwIfOtherPlatform: false);
+			TestRuntime.AssertSystemVersion (PlatformName.MacOSX, 10, 14, throwIfOtherPlatform: false);
```

Comment:
use `AssertXcode` (same reason as before) but add it to your `Setup` method (less code duplication)